code/rp_task.ipynb

```diff
@@ -67,24 +67,71 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 6,
-   "metadata": {
-    "lines_to_next_cell": 2
-   },
+   "execution_count": null,
+   "metadata": {},
    "outputs": [],
    "source": [
     "#******************************************\n",
     "# Data import\n",
     "#******************************************\n",
     "\n",
-    "with open('../datasets/RP_task_data_split.pickle', 'rb') as file_handle:\n",
-    "    read_data = pickle.load(file_handle)\n",
-    "    \n",
-    "vocab_dict = read_data['VocabDict']         # The vocabulary apart from the relative pronoun 'that' (with tag 'N' for\n",
-    "                                            # 'noun' and 'V' for 'verb' as the values of the dictionary).\n",
-    "data = read_data['Data']                    # The overall dataset as dictionary with labels [1,0] ([0,1]) as values.\n",
-    "training_data = read_data['TrainingData']   # List of sentences in the training subset.\n",
-    "testing_data = read_data['TestingData']     # List of sentences in the testing subset."
+    "# import train and test datasets: the data entries are all strings of the form 'label sentence' \n",
+    "# with the label in {0,1} and with the sentence of the form \"word1_POStag1 word2_POStag2 ...\"\n",
+    "\n",
+    "with open('../datasets/rp_train_data.txt') as f:\n",
+    "    training_data_raw = f.readlines()\n",
+    "\n",
+    "with open('../datasets/rp_test_data.txt') as f:\n",
+    "    testing_data_raw = f.readlines()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#***************************************************************\n",
+    "# Turn the raw input data into data structures convenient below\n",
+    "#***************************************************************\n",
+    "\n",
+    "vocab_dict = dict()     # dictionary to be filled with the vocabulary in the form { word : POStag }\n",
+    "data = dict()           # dictionary to be filled with all the data (from train and test subsets); entries of the \n",
+    "                        # form { sentence : label } with label encoding '1' as [1.0, 0.0] and '0' as [0.0, 1.0].\n",
+    "training_data = []      # list of sentences in the train dataset as strings \"word1 word2 ...\"\n",
+    "testing_data = []       # list of sentences in the test dataset as strings \"word1 word2 ...\"\n",
+    "\n",
+    "# Go through the train data\n",
+    "for sent in training_data_raw:\n",
+    "    words = sent[2:].split() \n",
+    "    sent_untagged = ''\n",
+    "    for word in words:\n",
+    "        if word != 'that': \n",
```

Comment:
If we add a tag for 'that' in the datasets, we will not need this special treatment here.